packages/flame/lib/src/components/core/component.dart

```diff
@@ -302,7 +302,7 @@ class Component {
   ///
   /// If [includeSelf] is set to true (default is false) then the component
   /// which the call is made for is also included in the search.
-  T? findParent<T extends Component>({bool includeSelf = false}) {
```

Comment:
how can your ancestor not be a component?

Reply:
If you check against a mixin for example, which I think is a pretty common usecase

Reply:
wouldn't the mixin be `on Component` or `on PositionComponent`?

Reply:
Not necessarily, see for example `PositionProvider` and all the other "provider" classes in that file,

Reply:
On the other hand, then that type is not really a `parent`, so I guess it is better to go through the `ancestors` method instead.